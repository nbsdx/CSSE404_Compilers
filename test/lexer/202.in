
/* More structured code tests. Intermixed comments and functions. Weird formatting */
// Random Comment.

class Class


{
    int func( int input ) {
        if( input <= 0 )
        {
            return 1
        }
        else
        {
            return input * func( input - 2 );
        }
    }

    public static 
    void main(


                String[] args) 
    {
        
        int var = func( 123 );

        if( ( var == 1 ) == true )
        {
            while( var > 1 )
            {
                System.out.println( var );
                var = var - 1;
            }
        }
    }
}
