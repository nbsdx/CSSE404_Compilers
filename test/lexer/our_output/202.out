ReservedWord, class
ID, Class
Delimiter, {
ReservedWord, int
ID, func
Delimiter, (
ReservedWord, int
ID, input
Delimiter, )
Delimiter, {
ReservedWord, if
Delimiter, (
ID, input
Operator, <=
Integer, 0
Delimiter, )
Delimiter, {
ReservedWord, return
Integer, 1
Delimiter, }
ReservedWord, else
Delimiter, {
ReservedWord, return
ID, input
Operator, *
ID, func
Delimiter, (
ID, input
Operator, -
Integer, 2
Delimiter, )
Delimiter, ;
Delimiter, }
Delimiter, }
ReservedWord, public
ReservedWord, static
ReservedWord, void
ReservedWord, main
Delimiter, (
ReservedWord, String
Delimiter, [
Delimiter, ]
ID, args
Delimiter, )
Delimiter, {
ReservedWord, int
ID, var
Delimiter, =
ID, func
Delimiter, (
Integer, 123
Delimiter, )
Delimiter, ;
ReservedWord, if
Delimiter, (
Delimiter, (
ID, var
Operator, ==
Integer, 1
Delimiter, )
Operator, ==
ReservedWord, true
Delimiter, )
Delimiter, {
ReservedWord, while
Delimiter, (
ID, var
Operator, >
Integer, 1
Delimiter, )
Delimiter, {
ReservedWord, System.out.println
Delimiter, (
ID, var
Delimiter, )
Delimiter, ;
ID, var
Delimiter, =
ID, var
Operator, -
Integer, 1
Delimiter, ;
Delimiter, }
Delimiter, }
Delimiter, }
Delimiter, }
